### ISSUE TITLE
brew install vim --with-python3

### ISSUE THREAD

**Issue by jwhite007** [Homebrew/homebrew#48925]
Upgrading vim after installed in this way leads to install of python2. "brew install vim --with-python3 --without-python" was needed.

**Comment by ilovezfs** [comment]
@mikemcquaid @jwhite007 I cannot reproduce this issue.

**Comment by DomT4** [comment]
We'll need full `brew gist-logs` here to take a look probably, but you may have overwritten them by doing the fresh install afterwards.

**Comment by jwhite007** [comment]
Thanks for the replies, guys.

I've uninstalled vim installed with "brew install vim --with-python3 --without-python" and reinstalled with "brew install vim --with-python3." This leads to an install of python which I don't want. Here are the logs...
https://gist.github.com/e2f0d7603ba0accd99ac. I then did a "brew uninstall python" followed by a "brew reinstall vim" which also leads to an install of python. Here are the logs... https://gist.github.com/4d8945c12f1c9c6698ce.

The only way that I can get "brew upgrade vim" and "brew reinstall vim" to not install python is with an initial install with "brew install vim --with-python3 --without-python". Here are the logs for that... https://gist.github.com/4b64fe5e2b1fce7237db

**Comment by jwhite007** [comment]
The problem has been solved. I'm not sure exactly which of the following two is responsible... Upon "brew cleanup," I was messaged that vim was not completely removed, that I had to issue "brew cleanup --force vim." I also had a sym link, "/usr/local/bin/python" to "/usr/local/bin/python3." I deleted the symlink. After doing both of these "brew reinstall vim" works as expected and does not attempt to install python. I assume the same will be true for "brew upgrade vim."
